network.go

```diff
@@ -55,22 +55,94 @@ type EndpointWalker func(ep Endpoint) bool
 
 type svcMap map[string]net.IP
 
+// IpamConf contains all the ipam related configurations for a network
+type IpamConf struct {
+	PreferredPool string
+	SubPool       string
+	Options       map[string]string // IPAM input options
+	IsV6          bool
+	Gateway       string
+	Addresses     map[string]string
```

Comment:
Similar to earlier comment. can you name is explicitly to reflect that fact that Addresses represent the secondary/auxilary addresses that user allocates to help the driver with its operation ?